packages/relay-runtime/mutations/RelayRecordSourceMutator.js

```diff
@@ -148,8 +149,19 @@ class RelayRecordSourceMutator {
   }
 
   copyFieldsFromRecord(record: Record, sinkID: DataID): void {
-    this.copyFields(RelayModernRecord.getDataID(record), sinkID);
```

Comment:
calling `copyFields` is not desirable because as the code is written, the sink gets overwritten with the base, then the mutated sink overwrites itself. This itself is a 2nd bug that is worthy of its own PR, but since this PR depends on fixing the bug, I kept them at 1 PR.

Reply:
this may be the same issue i'm experiencing here https://github.com/facebook/relay/pull/2489?

Reply:
sounds like it. those tests should pass in this PR. if they don't let me know.